plugins/module_utils/network/sonic/config/mclag/mclag.py

```diff
@@ -193,6 +200,64 @@ def _state_deleted(self, want, have):
                     commands = update_states(diff_want, "deleted")
         return commands, requests
 
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+        commands = []
+
+        diff1 = get_diff(have, want, TEST_KEYS, is_skeleton=True)
+        if diff1:
+            requests = self.get_delete_all_mclag_domain_request(have)
+            if len(requests) > 0:
+                send_requests(self._module, requests)
+            else:
+                commands = []
+
+        exist_mclag_facts = self.get_mclag_facts()
+        have_new = exist_mclag_facts
+        diff_new = get_diff(want, have_new)
+
+        if diff_new:
+            requests = self.get_create_mclag_request(want, diff_new)
+            if len(requests) > 0:
+                commands = update_states(diff, "replaced")
+        return commands, requests
+
+    def _state_overridden(self, want, have, diff):
+        """ The command generator when state is overridden
+        :param want: the desired configuration as a dictionary
+        :param obj_in_have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        diff1 = get_diff(have, want)
+        if not diff and not diff1:
+            return commands, requests
+
+        requests = self.get_delete_all_mclag_domain_request(have)
+        if len(requests) > 0:
+            send_requests(self._module, requests)
+        else:
+            commands = []
+
+        exist_mclag_facts = self.get_mclag_facts()
```

Comment:
Getting the facts is not needed. Either there is no action needed because no change has been requested (based on "diff"), or everything in "want" needs to be configured because a "delete all" has just been done.

Reply:
Facts recalculated to handle the case of change in domain id. please refer my previous comment.